### PR TITLE
Patterns Library: Start tracking calypso_pattern_library_view

### PR DESCRIPTION
**Pull request by olaseni**
see: pcYYuD-1hV-p2#calypso_pattern_library_view

This set of changes extend the `<PatternsPageViewTracker/>` component such that:

- `calypso_pattern_library_search` is no longer being tracked.
- `calypso_pattern_library_view`  is now being tracked

## Testing
- Navigate to any page in the `/patterns/*` route
- confirm the aforementioned events

### CODE REVIEW

File: client/my-sites/patterns/components/page-view-tracker.tsx [hunk 1]
```diff
@@ -38,19 +46,19 @@ export function PatternsPageViewTracker( { category, searchTerm }: PatternsPageV
 	}, [ category, isDevAccount, isLoggedIn ] );
```
Comment (fredrikekelund) [inline client/my-sites/patterns/components/page-view-tracker.tsx]: I think we could wrangle the updated properties for the `calypso_pattern_library_filter` event too in this PR.

File: client/my-sites/patterns/components/page-view-tracker.tsx [hunk 2]
```diff
@@ -4,16 +4,24 @@ import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
...
+export function PatternsPageViewTracker( {
+	category,
+	searchTerm,
+	type,
+	view,
+}: PatternsPageViewTrackerProps ) {
 	const isLoggedIn = useSelector( isUserLoggedIn );
 	const isDevAccount = useSelector( ( state ) => getUserSetting( state, 'is_dev_account' ) );
-	const [ debouncedSeachTerm, setDebouncedSearchTerm ] = useState( searchTerm );
+	const [ debouncedSearchTerm, setDebouncedSearchTerm ] = useState( searchTerm );
```
Comment (fredrikekelund) [inline client/my-sites/patterns/components/page-view-tracker.tsx:25]: 👍

File: client/my-sites/patterns/components/pattern-library/index.tsx [hunk 3]
```diff
@@ -103,17 +104,18 @@ export const PatternLibrary = ( {
 	const isLoggedIn = useSelector( isUserLoggedIn );
 	const isDevAccount = useSelector( ( state ) => getUserSetting( state, 'is_dev_account' ) );
 
-	const handleViewChange = ( view: 'grid' | 'list' ) => {
-		const currentView = isGridView ? 'grid' : 'list';
+	const type = patternTypeFilter === PatternTypeFilter.REGULAR ? 'pattern' : 'page-layout';
```
Comment (fredrikekelund) [inline client/my-sites/patterns/components/pattern-library/index.tsx]:
```suggestion
	const tracksPatternType = patternTypeFilter === PatternTypeFilter.REGULAR ? 'pattern' : 'page-layout';
```
It doesn't have to be this exact name, but seeing as this component already holds a fair amount of code, I think we should be more specific with the variable names.

File: client/my-sites/patterns/components/pattern-library/index.tsx [hunk 4]
```diff
@@ -103,17 +104,18 @@ export const PatternLibrary = ( {
...
 		}
 
 		recordTracksEvent( 'calypso_pattern_library_view_switch', {
 			category,
 			is_logged_in: isLoggedIn,
-			type: patternTypeFilter === PatternTypeFilter.REGULAR ? 'pattern' : 'page-layout',
+			type,
 			user_is_dev_account: isDevAccount ? '1' : '0',
 			view,
 		} );
```
Comment (fredrikekelund) [inline client/my-sites/patterns/components/pattern-library/index.tsx]: Maybe we should consider moving this into `PatternsPageViewTracker` now that we pass the `view` prop to that component?

**Review by fredrikekelund — APPROVED** [review]
The `calypso_pattern_library_view` works as expected 👍 I left a couple of comments that would be good to address before landing this, but seeing as I'm AFK tomorrow, I'll approve this straight away to unblock you

File: client/my-sites/patterns/components/page-view-tracker.tsx (hunk 1 above)
Comment (olaseni) [reply client/my-sites/patterns/components/page-view-tracker.tsx]: Good eye!

File: client/my-sites/patterns/components/pattern-library/index.tsx (hunk 3 above)
Comment (olaseni) [reply client/my-sites/patterns/components/pattern-library/index.tsx]: `patternType` sounds okay.

**Review by nightnei — CHANGES_REQUESTED** [review]
LGTM and works as expected, but as I see, now we fire the even `calypso_pattern_library_view ` two times:
Screenshot 2024-03-22 at 10 36 00

I double checked - before we had also two times... hm.

Maybe I am missing something, but I suppose that it's not expected behaviour. And to fix it we need to do boolean conversion here https://github.com/Automattic/wp-calypso/pull/88763/files#r1535527198

File: client/my-sites/patterns/components/page-view-tracker.tsx [hunk 5]
```diff
@@ -33,24 +42,25 @@ export function PatternsPageViewTracker( { category, searchTerm }: PatternsPageV
...
+			type: getTracksPatternType( patternTypeFilter ),
+			view,
+		} );
 
-		// We want to avoid resubmitting the `calypso_pattern_library_search` event whenever
+		// We want to avoid resubmitting the event whenever
 		// `category` changes, which is why we deliberately don't include it in the dependency array
 		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [ isDevAccount, isLoggedIn, debouncedSeachTerm ] );
+	}, [ isDevAccount, isLoggedIn, debouncedSearchTerm, patternTypeFilter, view ] );
```
Comment (nightnei) [inline client/my-sites/patterns/components/page-view-tracker.tsx:63]:
```suggestion
	}, [ !! isDevAccount, isLoggedIn, debouncedSearchTerm, patternTypeFilter, view ] );
```

File: client/my-sites/patterns/components/page-view-tracker.tsx [hunk 6]
```diff
@@ -4,16 +4,25 @@ import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
...
 
-export function PatternsPageViewTracker( { category, searchTerm }: PatternsPageViewTrackerProps ) {
+export function PatternsPageViewTracker( {
+	category,
+	searchTerm,
+	patternTypeFilter,
+	view,
+}: PatternsPageViewTrackerProps ) {
 	const isLoggedIn = useSelector( isUserLoggedIn );
 	const isDevAccount = useSelector( ( state ) => getUserSetting( state, 'is_dev_account' ) );
```
Comment (nightnei) [inline client/my-sites/patterns/components/page-view-tracker.tsx:24]:
```suggestion
	const isDevAccount = !! useSelector( ( state ) => getUserSetting( state, 'is_dev_account' ) );
```

**Review by nightnei — APPROVED** [review]
LGTM and works as expected, at least no new bug. But I noticed existed bug which we discussed with @olaseni to address in separate task. Details: p1711112211244959/1711102034.681499-slack-C06ELHR6L9J

**Pull request merged**